### PR TITLE
Update paperweight 1.20.4 - isOpaque does not exist (anymore)

### LINKED ISSUE #2607: ERROR]: An unexpected error while handling a FastAsyncWorldEdit command java.lang.NoSuchMethodError: 'boolean net.minecraft.world.level.block.state.IBlockData.isOpaque()'
### Server Implementation

Paper

### Server Version

1.20.4

### Describe the bug

当我在服务器使用//fixconnect时的报错

ERROR]: An unexpected error while handling a FastAsyncWorldEdit command
java.lang.NoSuchMethodError: 'boolean net.minecraft.world.level.block.state.IBlockData.isOpaque()'
        at com.sk89q.worldedit.bukkit.adapter.impl.fawe.v1_20_R3.PaperweightBlockMaterial.isOpaque(PaperweightBlockMaterial.java:78) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.arceon.core.command.FixConnectHandler.G(Unknown Source) ~[Arceon-0.4.4_1.16+.jar:?]
        at com.arceon.core.command.FixConnectHandler.fixconnect(Unknown Source) ~[Arceon-0.4.4_1.16+.jar:?]
        at com.arceon.core.command.FixConnectHandlerRegistration.A(Unknown Source) ~[Arceon-0.4.4_1.16+.jar:?]
        at org.enginehub.piston.CommandManager.execute(CommandManager.java:157) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.lambda$handleCommandOnCurrentThread$16(PlatformCommandManager.java:725) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.handleCommandTask(PlatformCommandManager.java:746) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.handleCommandOnCurrentThread(PlatformCommandManager.java:727) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.lambda$handleCommand$14(PlatformCommandManager.java:698) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.sk89q.worldedit.extension.platform.AbstractPlayerActor.lambda$runAction$1(AbstractPlayerActor.java:672) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.fastasyncworldedit.core.util.task.AsyncNotifyKeyedQueue.lambda$run$0(AsyncNotifyKeyedQueue.java:48) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at com.fastasyncworldedit.core.util.task.AsyncNotifyKeyedQueue.lambda$call$1(AsyncNotifyKeyedQueue.java:58) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at com.fastasyncworldedit.core.util.task.KeyQueuedExecutorService$KeyRunner.lambda$run$0(KeyQueuedExecutorService.java:150) ~[FastAsyncWorldEdit-Bukkit-2.9.0.jar:?]
        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?]
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]

### To Reproduce

//fixconnect

### Expected behaviour

当有调整链接状态

### Screenshots / Videos

_No response_

### Error log (if applicable)

_No response_

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/971b83c908234ac782fd304fcfb92746

### Fawe Version

FastAsyncWorldEdit-Bukkit-2.9.0.jar

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by PierreSchwang**
## Overview
Closes https://github.com/IntellectualSites/FastAsyncWorldEdit/pull/2604
Fixes #2607

Tested via #solidspread pattern and #liquid mask. Seems to work.

```[tasklist]
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

### CODE REVIEW

File: worldedit-bukkit/adapters/adapter-1_20_4/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_20_R3/PaperweightBlockMaterial.java [hunk 1]
```diff
@@ -75,7 +66,7 @@ public boolean isFullCube() {
 
     @Override
     public boolean isOpaque() {
-        return blockState.isOpaque();
+        return blockState.canOcclude();
```
Comment (dordsor21) [inline worldedit-bukkit/adapters/adapter-1_20_4/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_20_R3/PaperweightBlockMaterial.java:69]: `!canOcclude()`?

Comment (PierreSchwang) [reply worldedit-bukkit/adapters/adapter-1_20_4/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_20_R3/PaperweightBlockMaterial.java:69]: I've orientated myself on the 1.20.3 source if the BlockState#isOpaque method there 
image

Comment (PierreSchwang) [reply worldedit-bukkit/adapters/adapter-1_20_4/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_20_R3/PaperweightBlockMaterial.java:69]: Yeah, the same was present in `1.20.4-R0.1-20240106.182028-62` - which is kinda weird as therefor isSolid == isTranslucent == isOpaque lol

Comment (PierreSchwang) [reply worldedit-bukkit/adapters/adapter-1_20_4/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_20_R3/PaperweightBlockMaterial.java:69]: yeah, I'm dumb - didn't invert in isTranslucent

**Review by NotMyFault — APPROVED** [review]

**Review by SirYwell — APPROVED** [review]

**Comment by dordsor21** [comment]
Presumably there's nothing paper-specific here?

**Comment by PierreSchwang** [comment]
> Presumably there's nothing paper-specific here?

I'm not really testing with Spigot tbf - will take a look later via testing and looking through the source. 
I haven't seen any paper comments at least.

**Comment by PierreSchwang** [comment]
> Presumably there's nothing paper-specific here?

The vanilla mappings contain the methods, so I guess that's fairly safe to use
image
image
image

**Review by dordsor21 — APPROVED** [review]

**Pull request merged**